test/tests_for_integration/api_to_db_test.py

```diff
@@ -287,7 +368,238 @@ def test_get_admin_permission_success(ee2_port):
     assert ee2cli_write.get_admin_permission() == {"permission": "w"}
 
 
-def test_temporary_check_ws(ee2_port, ws_controller):
+######## run_job tests ########
+
+
+def _get_htc_mocks():
+    sub = create_autospec(htcondor.Submit, spec_set=True, instance=True)
+    schedd = create_autospec(htcondor.Schedd, spec_set=True, instance=True)
+    txn = create_autospec(htcondor.Transaction, spec_set=True, instance=True)
+    return sub, schedd, txn
+
+
+def _finish_htc_mocks(sub_init, schedd_init, sub, schedd, txn):
+    sub_init.return_value = sub
+    schedd_init.return_value = schedd
+    # mock context manager ops
+    schedd.transaction.return_value = txn
+    txn.__enter__.return_value = txn
+    return sub, schedd, txn
+
+
+def _check_htc_calls(sub_init, sub, schedd_init, schedd, txn, expected_sub):
+    sub_init.assert_called_once_with(expected_sub)
+    schedd_init.assert_called_once_with()
+    schedd.transaction.assert_called_once_with()
+    sub.queue.assert_called_once_with(txn, 1)
+
+
+def test_run_job(ee2_port, ws_controller, mongo_client):
+    """
+    A test of the run_job method.
+    """
+    # Set up workspace and objects
+    wsc = Workspace(ws_controller.get_url(), token=TOKEN_NO_ADMIN)
+    wsc.create_workspace({"workspace": "foo"})
+    wsc.save_objects(
+        {
+            "id": 1,
+            "objects": [
+                {"name": "one", "type": "Trivial.Object-1.0", "data": {}},
+                {"name": "two", "type": "Trivial.Object-1.0", "data": {}},
+            ],
+        }
+    )
```

Comment:
In a later PR I'll factor out the workspace setup, the condor expected sub creation, and the mongo job checking into functions for reuse with other tests in this module. Already done locally.